cirq-core/cirq/study/result.py

```diff
@@ -142,14 +142,16 @@ def dataframe_from_measurements(measurements: Mapping[str, np.ndarray]) -> pd.Da
         """
         # Convert to a DataFrame with columns as measurement keys, rows as
         # repetitions and a big endian integer for individual measurements.
-        converted_dict = {
-            key: [value.big_endian_bits_to_int(m_vals) for m_vals in val]
-            for key, val in measurements.items()
-        }
-        # Note that when a numpy array is produced from this data frame,
-        # Pandas will try to use np.int64 as dtype, but will upgrade to
-        # object if any value is too large to fit.
-        return pd.DataFrame(converted_dict, dtype=np.int64)
+        converted_dict = dict()
+        for key, bitstrings in measurements.items():
+            _, n = bitstrings.shape
+            i_type = object if n > 63 else np.int64
```

Comment:
nit: call this `dtype` instead
```suggestion
            dtype = object if n > 63 else np.int64
```